.github/workflows/ci-unit-tests-go-tip.yml

```diff
@@ -40,11 +41,12 @@ jobs:
       with:
         gh_token: ${{ secrets.GITHUB_TOKEN }}
 
-    - name: Clean Go cache for gotip
+    - name: Clean Go cache and rebuild stdlib for gotip
       run: |
         go clean -cache
         go clean -modcache
-
+        go install std
```

Comment:
this doesn't make sense to me. The fact that you have to do that indicates that we still have some cross-pollution happening between two Go versions. If `go clean` does not help we need to find a more reliable approach - once a gotip is installed no leftovers of the previous toolchain should be accessible. Maybe we need to be more aggressive about env variables, to ensure none of them point to the previous toolchain.